src/UI/Component/Dropzone/File/Factory.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace ILIAS\UI\Component\Dropzone\File;
+
+use ILIAS\UI\Component\Component;
+
+/**
+ * Interface Factory
+ *
+ * Describes a factory for file dropzones.
+ *
+ * @author  nmaerchy <nm@studer-raimann.ch>
+ * @author  Stefan Wanzenried <sw@studer-raimann.ch>
+ *
+ * @package ILIAS\UI\Component\Dropzone\File
+ */
+interface Factory {
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *      The standard dropzone is used to drop files dragged from outside
+	 *      the browser window. The dropped files are presented to the user and
+	 *      can be uploaded to the server. Standard dropzones CAN be used in
+	 *      forms.
+	 *   composition: >
+	 *      Standard dropzones consist of a visible area where files can
+	 *      be dropped. They MUST contain a message explaining that it is possible to
+	 *      drop files inside. The dropped files are presented to the user along
+	 *      with some button to start the upload process.
+	 *   effect: >
+	 *      A standard dropzone is highlighted when the user is dragging files
+	 *      over the dropzone. After dropping, the dropped files are presented
+	 *      to the user with some meta information of the files such the file name
+	 *      and file size.
+	 *   rivals:
+	 *      Rival 1: >
+	 *          A wrapper dropzone can hold other ILIAS UI components instead of
+	 *          a message.
+	 *
+	 * rules:
+	 *   usage:
+	 *     1: A page SHOULD contain only one standard dropzone.
+	 *     2: Standard dropzones MUST contain a message.
+	 *     3: >
+	 *        Standard dropzones MUST offer the possibility to select files
+	 *        manually from the computer.
+	 *     4: >
+	 *        The upload button MUST be disabled if there are no files
+	 *        to be uploaded.
+	 *   responsiveness:
+	 *     1: Standard dropzones SHOULD have a static height.
+	 *
+	 * ---
+	 *
+	 * @param string $url The url where the dropped files are being uploaded
+	 * @return \ILIAS\UI\Component\Dropzone\File\Standard
+	 */
+	public function standard($url);
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *      A wrapper dropzone is used to display other ILIAS UI components
+	 *      inside it. In contrast to the standard dropzone, the wrapper
+	 *      dropzone is not visible by default. Only the wrapped components are
+	 *      visible. Any wrapper dropzone gets highlighted once the user is dragging
+	 *      files over the browser window. Thus, a user needs to have the knowledge
+	 *      that there are wrapper dropzones present.
```

Comment:
Add some additional description of where they can be used in ILIAS concretely:
"They can be introduced to offer additional approaches to complete some workflow more conveniently. Especially in situation where space is scarce suche as appointments in the calendar".